pkg/runctl/run_manager.go

```diff
@@ -127,31 +127,46 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 	}
 
 	// configure service account in run namespace
+	err = c.setupServiceAccount(runNamespace, pipelineCloneSecretName, imagePullSecrets)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *runManager) setupServiceAccount(runNamespace string, pipelineCloneSecretName string, imagePullSecrets []string) error {
 	accountManager := k8s.NewServiceAccountManager(c.factory, runNamespace)
 	serviceAccount, err := accountManager.CreateServiceAccount(serviceAccountName, pipelineCloneSecretName, imagePullSecrets)
 	if err != nil {
 		if !k8serrors.IsAlreadyExists(err) {
 			return errors.Wrapf(err, "failed to create service account %q", serviceAccountName)
 		}
 
-		serviceAccount, err = accountManager.GetServiceAccount(serviceAccountName)
-		if err != nil {
-			return errors.Wrapf(err, "failed to get service account %q", serviceAccountName)
-		}
-		if pipelineCloneSecretName != "" {
-			serviceAccount.AttachSecrets(pipelineCloneSecretName)
-		}
-		serviceAccount.AttachImagePullSecrets(imagePullSecrets...)
-		err = serviceAccount.Update()
-		if err != nil {
-			return errors.Wrapf(err, "failed to update service account %q", serviceAccountName)
+		// service account exists already, so we need to attach secrets to it
+		for { // retry loop
+			serviceAccount, err = accountManager.GetServiceAccount(serviceAccountName)
+			if err != nil {
+				return errors.Wrapf(err, "failed to get service account %q", serviceAccountName)
+			}
+			if pipelineCloneSecretName != "" {
+				serviceAccount.AttachSecrets(pipelineCloneSecretName)
+			}
+			serviceAccount.AttachImagePullSecrets(imagePullSecrets...)
+			err = serviceAccount.Update()
+			if err == nil {
+				break // ...the retry loop
+			}
+			if !k8serrors.IsConflict(err) {
+				return errors.Wrapf(err, "failed to update service account %q", serviceAccountName)
+			}
```

Comment:
A comment could make it more readable
```suggestion
			}
			// conflict - object changed, retrying again with the latest version
```

Reply:
Fixed by ee32ee5a.